packages/neutrino/Neutrino.js

```diff
@@ -205,39 +138,15 @@ module.exports = class Neutrino {
     this.outputHandlers.set(name, handler);
   }
 
-  use(middleware, options) {
-    if (typeof middleware === 'function') {
-      // If middleware is a function, invoke it with the provided options
-      middleware(this, options);
-    } else if (typeof middleware === 'string') {
-      // If middleware is a string, it's a module to require.
-      // Require it, then run the results back through .use()
-      // with the provided options
-      this.use(requireFromRoot(middleware, this.options.root), options);
-    } else if (Array.isArray(middleware)) {
-      // If middleware is an array, it's a pair of some other
-      // middleware type and options
-      this.use(...middleware);
-    } else if (isPlainObject(middleware)) {
-      if ('env' in middleware) {
-        throw new ConfigurationError(
-          'Using "env" in middleware has been removed. Apply middleware conditionally instead.'
-        );
-      }
-      // If middleware is an object, it could contain other middleware in
-      // its "use" property. Run every item in "use" prop back through .use(),
-      // plus set any options.
-      if (middleware.options) {
-        this.options = this.mergeOptions(this.options, middleware.options);
-      }
+  use(middleware) {
+    if (!middleware) {
+      return;
+    }
 
-      if (middleware.use) {
-        if (Array.isArray(middleware.use)) {
-          middleware.use.map(usage => this.use(usage));
-        } else {
-          this.use(middleware.use);
-        }
-      }
+    if (typeof middleware !== 'function') {
+      throw new ConfigurationError('Middleware must be a function');
```

Comment:
If I try running `yarn test` without first changing Treeherder's `.neutrinorc.js`, I get:

```
treeherder $ yarn test
yarn run v1.13.0
$ node ./node_modules/jest/bin/jest

An error occurred when loading the Neutrino configuration.

ConfigurationError: Middleware must be a function
    at Neutrino.use (C:\Users\Ed\src\neutrino\packages\neutrino\Neutrino.js:147:13)
    at use.forEach.use (C:\Users\Ed\src\neutrino\packages\neutrino\index.js:42:37)
    at Array.forEach (<anonymous>)
    at module.exports (C:\Users\Ed\src\neutrino\packages\neutrino\index.js:42:13)
    at Object.<anonymous> (C:\Users\Ed\src\treeherder\jest.config.js:5:18)
    at Module._compile (internal/modules/cjs/loader.js:734:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:745:10)
    at Module.load (internal/modules/cjs/loader.js:626:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:566:12)
    at Function.Module._load (internal/modules/cjs/loader.js:558:3)
```

This is better than a more obscure TypeError or similar, however I feel that since this is likely the first error people see upon updating to Neutrino 9 (eg in CI when a bot opens an update PR), it might be worth making the error message slightly longer/clearer (eg "As of Neutrino 9, middleware must now be passed as a function and not an array or string"), and linking to the migration guide/migrate tool? 

Also, would it be possible to have the error mention where the invalid usage occurred? eg to differentiate top-level `.neutrinorc.js` vs one of their third-party presets not having been updated to support Neutrino 9?

Relatedly, if I put just `react` and not `react()` inside the `.neutrinorc.js` `use` array, at the moment I don't get a Neutrino error message and instead an obscure webpack message. I feel that an easy/common mistake for people to make, will be confusing the "middleware that takes options" and "middleware that does not" forms. 

As such, perhaps Neutrino should help catch this case? eg by either (or both of):
* verifying the passed middleware takes one and only one argument (using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/length)
* checking the passed middleware doesn't return a function of it's own